example_python_travis.yml

```diff
@@ -13,29 +14,32 @@ addons:
 
 env:
   global:
+    # Doctr deploy key for <org>/<repo>
     - secure: "<docs build key>"
+    # You only need to specify these if they are different from these values
+    - DOCS_FOLDER: docs
+    - DOCS_REQUIREMENTS: docs-requirements.txt
+    # Assign this variable if you want a version menu in your docs
+    - DOCTR_VERSIONS_MENU=1
+    # Assign this variable to include folders that start with an underscore in
+    # your gh-pages site. This is already done when using DOCTR_VERSIONS_MENU
+    # so you don't need to set both.
+    - NO_JEKYLL=1
 
-matrix:
-  include:
-    - name: flake8 linting
-      python: 3.6
-      env: LINT_PYTHON=pytmc
-    - python: 3.6
-      env: BUILD_DOCS=1
-    - python: 3.7
-
-install:
-  - git clone --depth 1 git://github.com/pcdshub/pcds-ci-helpers.git
-  - source pcds-ci-helpers/travis/init.sh
-
-  - pip install -r requirements.txt
-  - pip install -r dev-requirements.txt
- 
+# Also for X11:
 before_script:
   # Run the window manager
   - "herbstluftwm &"
   - sleep 1
 
-script:
-  - coverage run run_tests.py
-  - coverage report -m 
+stages:
+  - test
+  - name: deploy
+    if: (branch = master OR tag IS present) AND type != pull_request
```

Comment:
I like that this `stage / if` sets up conditions so that we don't have to in each script/configuration.